plugins/logitech-tap/fu-logitech-tap-sensor-device.c

```diff
@@ -90,16 +91,15 @@ static gboolean
 fu_logitech_tap_sensor_device_enable_tde(FuDevice *device, GError **error)
 {
 	FuLogitechTapSensorDevice *self = FU_LOGITECH_TAP_SENSOR_DEVICE(device);
-	guint8 buf[HID_SET_DATA_LEN] = {
-	    kHidMcuTdeReportId,
-	    kHidMcuTdeModeSelector,
-	    kHidMcuTdeModeEnable,
-	    0,
-	    0,
-	};
+	g_autoptr(FuStructLogitechTapSensorHidReq) st = fu_struct_logitech_tap_sensor_hid_req_new();
+	fu_struct_logitech_tap_sensor_hid_req_set_cmd(st,
+						      FU_STRUCT_LOGITECH_TAP_TOUCH_HID_SET_CMD_TDE);
+	fu_struct_logitech_tap_sensor_hid_req_set_payload_byte1(st, kHidMcuTdeModeSelector);
```

Comment:
I think moving kHidMcuTde* to rs format will give quite a bit of type safety too.